src/main/java/io/fabric8/elasticsearch/plugin/kibana/KibanaSeed.java

```diff
@@ -121,8 +143,85 @@ public static void setDashboards(String user, Set<String> projects, Set<String>
 				if ( common.size() > 0 )
 					setDefaultIndex(user, common.get(0), esClient, kibanaIndex, kibanaVersion);
 			}
+		}
+		
+		if ( changed )
+			refreshKibanaUser(user, kibanaIndex, esClient);
+	}
+	
+	private static void refreshKibanaUser(String username, String kibanaIndex, Client esClient) {
```

Comment:
do we need to synchronize this or impl state and atomics for multi-threaded access?